crates/core_simd/src/vector.rs

```diff
@@ -311,6 +312,57 @@ where
         unsafe { self.store(slice.as_mut_ptr().cast()) }
     }
 
+    #[must_use]
+    #[inline]
+    pub fn masked_load_or(slice: &[T], or: Self) -> Self
+    where
+        Mask<<T as SimdElement>::Mask, N>: ToBitMask + ToBitMaskArray,
+    {
+        Self::masked_load_select(slice, Mask::splat(true), or)
+    }
```

Comment:
I don't think this function does anything?

Reply:
I suppose it's masking the slice length, but I'm not sure I would even call this a masked load.  Also, I'm not sure this is the best way to load from slices on most architectures.

Reply:
It's supposed to enable us to write extremely nice loops like this:

```rust
let mut accum = Simd::<u8, N>::default();
for i in (0..data.len()).step_by(N) {
    accum ^= Simd::masked_load_or(&data[i..], Simd::default());
}
```

No epilogues or scalar fallbacks needed. This could be even shorter since `SimdElement` implies `Default`?

But it's only a reality on AVX-512 for now on account of the codegen

Reply:
I would at least rename it--it might be implemented via masked loads, but API-wise I would consider it something else entirely.  This is perhaps `Simd::from_slice`.

Reply:
After the rename -- I don't like that this function seems more discoverable than from_slice. Users should prefer `from_slice` if it fits their use-case. This isn't a problem with gather/scatter because that's a specific name with a more complex signature, it's hard to use it by accident.